domain/modelmigration/state/state.go

```diff
@@ -66,3 +68,36 @@ FROM model`, ModelInfo{})
 
 	return result.ControllerUUID, nil
 }
+
+// GetAllInstanceIDs returns all instance IDs from the current model as
+// juju/collections set.
+func (s *State) GetAllInstanceIDs(ctx context.Context) (set.Strings, error) {
+
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Errorf("cannot get database to retrieve instance IDs: %w", err)
+	}
+
+	query := `
+SELECT &instanceID.instance_id
+FROM   machine_cloud_instance`
```

Comment:
is this a convention that we have been following aligning the from and select? Just asking so I am aware of the pattern.